packages/payment-processor/src/payment/utils.ts

```diff
@@ -420,3 +421,31 @@ export async function revokeErc20Approval(
   const tx = await signer.sendTransaction(preparedTx);
   return tx;
 }
+
+/**
+ * Format a request we wish to build a payment for.
+ * If the request does not use the meta-pn, it returns it as is.
+ * Otherwise, returns the request formatted with the pn of interest
+ */
+export function getFormattedRequest({
```

Comment:
```suggestion
export function flattenRequestByPnId({
```
Or something more explicit than "format" which looks like cosmetic convenience
```suggestion
export function unMetaRequestByPnId({
```

Reply:
```suggestion
export function decodeRequestMetaPn({
```